runtime/syntax/nftables.yaml

```diff
@@ -1,19 +1,23 @@
 filetype: nftables
 
 detect:
-    filename: "nftables.conf$"
-    header: "^(#!.*/(env +)?nft( |$)|flush +ruleset)"
-    
+   filename: "(nftables\\.(conf|rules)$|nftables(\\.rules)?\\.d/)"
+   header: "^(#!.*/(env +)?nft( |$)|flush +ruleset)"
+
 rules:
     - type: "\\b(chain|counter|map|rule|ruleset|set|table)\\b"
-    - type: "\\b(ether|icmp|icmpv6|icmpx|inet|ip|ip6|ipv4|ipv6|tcp|udp)\\b"
-    - special: "\\b(elements|hook|policy|priority|type)\\b"
-    - identifier: "\\b(ct|iif|iifname|meta|oif|oifname|th)\\b"
-    - statement: "\\b(accept|drop|goto|jump|log|masquerade|reject)\\b"
+    - type: "\\b(ether|inet|i(cm)?p(x|(v?(4|6))?)|tcp|udp)\\b"
+    - special: "\\b(elements|hook|policy|priority|type|state)\\b"
+    - identifier: "\\b(ct|iif|iifname|meta|oif|oifname|th|dport|sport|saddr|daddr|l4proto)\\b"
+    - statement: "\\b(accept|drop|goto|jump|log|masquerade|reject|limit|queue)\\b"
     - preproc: "\\b(add|define|flush|include|delete)\\b"
-    - symbol: "[-=/:;,@]"
-    - symbol.operator: "[<>.&|^!]|\\b(and|ge|gt|le|lt|or|xor)\\b"
-    - constant.string: '([\"]{1})(.*)([\"]{1})'
+    - symbol.operator: "[<>.&|^!=:;,@]|\\b(and|ge|gt|le|lt|or|xor)\\b"
+    - constant.string: '([\+\-\*/][\"]{1})(.*)([\"]{1})'
+    - constant.string:
+         start: "\""
+         end: "\""
+      # Integer Constants
+    - constant.number: "([-+]?[1-9]\\d*)"
```

Comment:
You ignored the `0` as integer.
Maybe directly `\\b((0x)?[0-9a-fA-F]+)\\b` to support dec and hex at once? This will fix the issue in which e.g. the `6` in `ipv6` will be marked as integer. Additionally we then have MAC address and IPv6 address highlighting.

Reply:
okay. what means the `\\b` in the regex? i saw this now multiple times in your regexes.

Reply:
`\b` matches at word boundary, while `\\b` just includes one more slash to escape the second one.

Reply:
okay thank you for the explanation